texar/torch/modules/decoders/decoder_base.py

```diff
@@ -408,20 +408,23 @@ def dynamic_decode(self, helper: Helper, inputs: Optional[torch.Tensor],
                (max_decoding_length is None or time < max_decoding_length)):
 
             next_outputs, decoder_state = \
-                self.compute_output_state(helper, time, step_inputs, state)
+                self.step(helper, time, step_inputs, state)
 
             if max_decoding_length is not None and \
                     time + 1 == max_decoding_length:
                 # Maximum decoding length reached, mark all batches as finished.
                 # This requires special handling because performing lookup on
                 # position embeddings with `time + 1` may result in IndexError.
-                decoder_finished = torch.tensor(1, dtype=torch_bool)
+                decoder_finished = torch.tensor(1, dtype=torch_bool,
+                                                device=finished.device)
                 # Since `next_inputs` will not be used, simply create a null
                 # tensor.
                 next_inputs = torch.empty(0)
             else:
-                next_inputs, decoder_finished = \
-                    self.compute_next_input(helper, time, next_outputs)
+                logits = next_outputs.logits  # type: ignore
+                sample_id = next_outputs.sample_id  # type: ignore
```

Comment:
Why do we need `type: ignore` here?

Reply:
`next_outputs` is an `Output` object, and it has no attributes `logits` and `sample_id`.